### PR TITLE
Fixes storageCluster status when Alerts are firing

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshot from 2023-05-22 14-04-16

### CODE REVIEW

File: packages/shared/models/common.ts [hunk 1]
```diff
@@ -248,3 +248,16 @@ export const DaemonSetModel: K8sKind = {
...
+  labelPlural: 'Storage Clusters',
+  apiVersion: 'v1',
+  apiGroup: 'ocs.openshift.io',
+  plural: 'storageclusters',
+  abbr: 'OCS',
+  namespaced: true,
+  kind: 'StorageCluster',
+  id: 'ocscluster',
+  crd: true,
+};
```
Comment (SanjalKatiyar) [inline packages/shared/models/common.ts:263]: we have "OCSStorageClusterModel" in "shared" and "StorageClusterModel" in "ocs"...

File: packages/shared/topology/utils/AlertFilters.ts [hunk 2]
```diff
@@ -10,6 +10,29 @@ export const commonFilter =
...
+  _.get(alert, 'annotations.storage_type') === 'NooBaa';
+
+export const filterNooBaaAlerts = (alerts: Alert[]): Alert[] =>
+  alerts?.filter(noobaaFilter);
+
+const rgwFilter = (alert: Alert) => alert?.annotations?.storage_type === 'RGW';
+
+export const filterRGWAlerts = (alerts: Alert[]): Alert[] =>
+  alerts?.filter(rgwFilter);
+
```
Comment (SanjalKatiyar) [inline packages/shared/topology/utils/AlertFilters.ts:35]: we should remove these from "ocs/utils/metrics.ts" and let "ocs" import these from "shared" as well..

Comment (SanjalKatiyar) [reply packages/shared/topology/utils/AlertFilters.ts:35]: if it is shared between "topology" and "ocs", we can keep them in general purpose utils...

File: packages/shared/models/common.ts (hunk 1 above)
Comment (bipuladh) [reply packages/shared/models/common.ts:263]: Yeah, looks like all these models belong in `shared`. The reason being our codebase is completely ODF centric and all our packages at some point will need these models. I could make the changes in this PR but that would make a huge backport PR so I will create an issue to track this.

File: packages/shared/topology/utils/AlertFilters.ts [hunk 3]
```diff
@@ -23,6 +46,10 @@ export const filterRelevantAlerts = (
   if (_.isEqual(resourceModel, NodeModel)) {
     filter = nodeFilter(resourceName);
   }
+  if (_.isEqual(StorageClusterModel, resourceModel)) {
+    filter = (alert: Alert) =>
+      rgwFilter(alert) || noobaaFilter(alert) || cephFilter(alert);
```
Comment (SanjalKatiyar) [inline packages/shared/topology/utils/AlertFilters.ts:51]: no need for this here ??
```suggestion
    filter = (alert: Alert) =>
      rgwFilter(alert) || noobaaFilter(alert) || cephFilter(alert) || commonFilter(alert);
```

Comment (SanjalKatiyar) [inline packages/shared/topology/utils/AlertFilters.ts:51]: no need for " || commonFilter" here ??

**Comment by SanjalKatiyar** [comment]
@bipuladh PR needs rebase...

**Comment by SanjalKatiyar** [comment]
@bipuladh we need manual backport...

**Pull request merged**